### ISSUE TITLE
HTML entities like '>' are not escaped when saving / parsed correctly when reading from markdown

### ISSUE THREAD

**Issue by lokimckay** [tinacms/tinacms#3619]
### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

I am trying to use a custom-defined MDX element that has a "content" rich-text field

Within that rich text field I want to save some content that directs my readers to a file path like this:
```
Navigate to this file:
path -> to -> file
```

Tina spits the following error at me about the ">" character when the markdown file is saved:

```graphql
Could not parse expression with acorn: Unexpected token `>`. Did you mean `&gt;` or `{">"}`?
```

I noticed a similar error is also reported for the less-than character '<'

### Reproduction

[https://tina-gql-playground.vercel.app/rich-text?markdownCode...](https://tina-gql-playground.vercel.app/rich-text?markdownCode=---%0A---%0A%0A%23%23+Welcome%0A%0ALearn+about+working+with+%5Brich+text%5D%28https%3A%2F%2Ftina.io%2Fdocs%2Fediting%2Fmdx%2F%29%21%0A%0A%3CCallout+content%3D%7B%3C%3E%3Cp%3EHello+there%3C%2Fp%3E%3C%2F%3E%7D+%2F%3E%0A%0A&schemaCode=import+%7B+defineSchema%2C+defineConfig+%7D+from+%27tinacms%27%0A%0Aconst+schema+%3D+defineSchema%28%7B%0A++collections%3A+%5B%7B%0A++++label%3A+%22Post%22%2C%0A++++name%3A+%22post%22%2C%0A++++path%3A+%22posts%22%2C%0A++++fields%3A+%5B%7B%0A++++++label%3A+%22Body%22%2C%0A++++++name%3A+%22body%22%2C%0A++++++isBody%3A+true%2C%0A++++++type%3A+%22rich-text%22%2C%0A++++++templates%3A+%5B%0A++++++++%7B%0A++++++++++name%3A+%22Callout%22%2C%0A++++++++++label%3A+%22Callout%22%2C%0A++++++++++fields%3A+%5B%7B%0A++++++++++++name%3A+%22content%22%2C%0A++++++++++++label%3A+%22Content%22%2C%0A++++++++++++type%3A+%22rich-text%22%0A++++++++++%7D%0A++++++++%5D%7D%0A++++++%5D%0A++++%7D%5D%0A++%7D%5D%0A%7D%29%0Aexport+default+defineConfig%28%7B+schema+%7D%29%0A&reactCode=import+*+as+React+from+%27react%27%0Aimport+%7B+useTina+%7D+from+%27tinacms%2Fdist%2Freact%27%0Aimport+%7B+useCMS+%7D+from+%27tinacms%27%0Aimport+%7B+TinaMarkdown+%7D+from+%22tinacms%2Fdist%2Frich-text%22%3B%0A%0Aconst+Callout+%3D+%28%7B+content+%7D%29+%3D%3E+%7B%0A++if+%28%21content%29+return+null%3B%0A++return+%28%3Cdiv+className%3D%22callout%22%3E%3CTinaMarkdown+content%3D%7Bcontent%7D+%2F%3E%3C%2Fdiv%3E%29%0A%7D%0A%0A%2F%2F+Note%3A+In+most+cases%2C+this+query+would+be%0A%2F%2F+autogenerated+with+%60client.queries.%3CCollection%3E%60%0A%2F%2F+https%3A%2F%2Ftina.io%2Fdocs%2Ffeatures%2Fdata-fetching%2F%0Aconst+QUERY+%3D+%60query+%7B%0A++post%28relativePath%3A+%22hello-world.md%22%29+%7B%0A++++body%0A++%7D%0A%7D%60%0A%0Aexport+default+function+Page%28props%29+%7B%0A++const+cms+%3D+useCMS%28%29%0A++React.useMemo%28%28%29+%3D%3E+%7B%0A++++cms.flags.set%28%27rich-text-alt%27%2C+true%29%0A++%7D%2C+%5Bcms%5D%29%0A++const+%7Bdata%2C+isLoading%7D+%3D+useTina%28%7B+query%3A+QUERY%2C%0A++++variables%3A+%7B%7D%2C+data%3A+props.data%0A++%7D%29%0A%0A++if%28isLoading%29+%7B%0A++++return+%3Cdiv%3ELoading...%3C%2Fdiv%3E%0A++%7D%0A%0A++return+%28%0A++++%3Cdiv+className%3D%22bg-white%22%3E%0A++++++%3Cdiv+className%3D%22max-w-7xl+mx-auto+py-6+px-4+sm%3Apx-6+lg%3Apx-8%22%3E%0A++++++++%3Cdiv+className%3D%22prose%22%3E%0A++++++++++%3CTinaMarkdown%0A++++++++++++content%3D%7Bdata.post.body%7D%0A++++++++++++components%3D%7B%7BCallout%7D%7D%0A++++++++++%2F%3E%0A++++++++%3C%2Fdiv%3E%0A++++++%3C%2Fdiv%3E%0A++++%3C%2Fdiv%3E%0A++%29%0A%7D)

### Steps to reproduce

1. Open reproduction link
2. Open the Tina sidebar
3. Enter the custom "Callout" element settings
4. Add the following text to the "Content" field
```
HTML entities -> are not -> escaped!
```
5. Click save
6. Observe error reported

### Workaround

[Repro link](https://tina-gql-playground.vercel.app/rich-text?markdownCode=---%0A---%0A%0A%23%23+Welcome%0A%0ALearn+about+working+with+%5Brich+text%5D%28https%3A%2F%2Ftina.io%2Fdocs%2Fediting%2Fmdx%2F%29%21%0A%0A%3CCallout+content%3D%7B%3C%3E%3Cp%3EHello+there%3C%2Fp%3E%3C%2F%3E%7D+%2F%3E%0A%0A&schemaCode=import+%7B+defineSchema%2C+defineConfig+%7D+from+%27tinacms%27%0A%0Aconst+schema+%3D+defineSchema%28%7B%0A++collections%3A+%5B%7B%0A++++label%3A+%22Post%22%2C%0A++++name%3A+%22post%22%2C%0A++++path%3A+%22posts%22%2C%0A++++fields%3A+%5B%7B%0A++++++label%3A+%22Body%22%2C%0A++++++name%3A+%22body%22%2C%0A++++++isBody%3A+true%2C%0A++++++type%3A+%22rich-text%22%2C%0A++++++templates%3A+%5B%0A++++++++%7B%0A++++++++++name%3A+%22Callout%22%2C%0A++++++++++label%3A+%22Callout%22%2C%0A++++++++++fields%3A+%5B%7B%0A++++++++++++name%3A+%22content%22%2C%0A++++++++++++label%3A+%22Content%22%2C%0A++++++++++++type%3A+%22string%22%2C%0A++++++++++++ui%3A+%7B%0A++++++++++++++component%3A+%22textarea%22%0A++++++++++++%7D%0A++++++++++%7D%0A++++++++%5D%7D%0A++++++%5D%0A++++%7D%5D%0A++%7D%5D%0A%7D%29%0Aexport+default+defineConfig%28%7B+schema+%7D%29%0A&reactCode=import+*+as+React+from+%27react%27%0Aimport+%7B+useTina+%7D+from+%27tinacms%2Fdist%2Freact%27%0Aimport+%7B+useCMS+%7D+from+%27tinacms%27%0Aimport+%7B+TinaMarkdown+%7D+from+%22tinacms%2Fdist%2Frich-text%22%3B%0A%0Aconst+Callout+%3D+%28%7B+content+%7D%29+%3D%3E+%7B%0A++if+%28%21content%29+return+null%3B%0A++return+%28%3Cdiv+className%3D%22callout%22%3E%7Bcontent%7D%3C%2Fdiv%3E%29%0A%7D%0A%0A%2F%2F+Note%3A+In+most+cases%2C+this+query+would+be%0A%2F%2F+autogenerated+with+%60client.queries.%3CCollection%3E%60%0A%2F%2F+https%3A%2F%2Ftina.io%2Fdocs%2Ffeatures%2Fdata-fetching%2F%0Aconst+QUERY+%3D+%60query+%7B%0A++post%28relativePath%3A+%22hello-world.md%22%29+%7B%0A++++body%0A++%7D%0A%7D%60%0A%0Aexport+default+function+Page%28props%29+%7B%0A++const+cms+%3D+useCMS%28%29%0A++React.useMemo%28%28%29+%3D%3E+%7B%0A++++cms.flags.set%28%27rich-text-alt%27%2C+true%29%0A++%7D%2C+%5Bcms%5D%29%0A++const+%7Bdata%2C+isLoading%7D+%3D+useTina%28%7B+query%3A+QUERY%2C%0A++++variables%3A+%7B%7D%2C+data%3A+props.data%0A++%7D%29%0A%0A++if%28isLoading%29+%7B%0A++++return+%3Cdiv%3ELoading...%3C%2Fdiv%3E%0A++%7D%0A%0A++return+%28%0A++++%3Cdiv+className%3D%22bg-white%22%3E%0A++++++%3Cdiv+className%3D%22max-w-7xl+mx-auto+py-6+px-4+sm%3Apx-6+lg%3Apx-8%22%3E%0A++++++++%3Cdiv+className%3D%22prose%22%3E%0A++++++++++%3CTinaMarkdown%0A++++++++++++content%3D%7Bdata.post.body%7D%0A++++++++++++components%3D%7B%7BCallout%7D%7D%0A++++++++++%2F%3E%0A++++++++%3C%2Fdiv%3E%0A++++++%3C%2Fdiv%3E%0A++++%3C%2Fdiv%3E%0A++%29%0A%7D)
If you can sacrifice rich-text for a multiline string instead, this schema works with the `>` character

```js
{
  name: "Callout",
  label: "Callout",
  fields: [
    {
      name: "content",
      label: "Content",
      type: "string",
      component: "textarea"
    },
  ],
},
```

### System Info

```shell
Not relevant - tina playground environment
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

**Comment by lokimckay** [comment]
Additional info: I tried to workaround this problem using the [custom shortcode syntax instead](https://tina.io/docs/reference/types/rich-text/#custom-shortcode-syntax)

But the ">" character is also not saved correctly with that method:

Templates schema:
```js
{
  name: "Callout",
  label: "Callout",
  match: {
    start: "{{",
    end: "}}"
  },
  fields: [
    {
      name: "content",
      label: "Content",
      type: "string",
      component: "textarea"
    },
  ],
},
```

Input in TinaCMS sidebar content field
```
To add javascript on all pages on Squarespace, go to `Settings -> Advanced -> Code Injection`
```

Result in markdown after pressing save:
```mdx
{{ Callout content="To add javascript on all pages on Squarespace, go to `Settings - }}
```

**Comment by logan-anderson** [comment]
@lokimckay We did some updates around this. Mind updating to the latest version and see if it is still an issue?

**Comment by lokimckay** [comment]
Thanks @logan-anderson! can confirm this is resolved for me after updating to `"tinacms": "^1.2.2",`

**Comment by lokimckay** [comment]
Sorry I spoke too soon, this issue is only partially resolved.

When using version `1.2.2` the `>` character is correctly saved into markdown via the tina sidebar:
![image](https://user-images.githubusercontent.com/14341953/222884563-0144e874-90bc-4d6b-b09d-abe33016a7ec.png)

However it seems like Tina still can't parse the `>` character and render it properly:
![image](https://user-images.githubusercontent.com/14341953/222884586-c6aa28a1-1280-41d9-aa4e-975be1b07448.png)
